### PR TITLE
fix #23: refactor reserved/restricted word validation

### LINKED ISSUE #23: Port the changes in https://github.com/shapesecurity/shift-validator-js/pull/8

### PR DESCRIPTION
**Pull request by ikarienator**

### CODE REVIEW

**Comment by michaelficarra** [comment]
You've not consolidated the restricted word checking at all, just reserved word checks.

**Comment by ikarienator** [comment]
I'm not sure what you meant. Restricted checking was in place before. Your change was just a simplification on the implementation.

**Comment by michaelficarra** [comment]
Ping @ikarienator. Also, please rebase.

**Comment by ikarienator** [comment]
The simplification comes with a cost. Not going to add it.

File: src/main/java/com/shapesecurity/shift/validator/ValidationContext.java [hunk 1]
```diff
@@ -67,82 +71,229 @@ private ValidationContext(@NotNull ConcatList<ValidationError> freeBreakStatemen
...
-  public ValidationContext addFreeContinueStatement(@NotNull ValidationError statement) {
+  public ValidationContext addFreeContinueStatement(@NotNull ContinueStatement statement) {
     return new ValidationContext(
-        this.freeBreakStatements, this.freeContinueStatements.append(
-        ConcatList.single(
-            statement)), this.usedLabelNames, this.freeJumpTargets, this.freeReturnStatements, this.errors,
-        this.strictErrors);
+        this.freeBreakStatements,
+        this.freeContinueStatements.append1(new ValidationError(statement,
+            "Continue statement must be inside a recursive loop")),
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/shift/validator/ValidationContext.java]: Can you change this "recursive loop" message? It must be in an iteration statement.

**Pull request merged**